src/main/java/duke/Duke.java

```diff
@@ -0,0 +1,95 @@
+package duke;
+
+import java.io.FileWriter;
+import java.io.IOException;
+import java.util.Scanner;
+
+
+/**
+ * Duke is an application to help store and track tasks.
+ * Users are able to add and delete various types of tasks such as deadlines, events, and things to do.
+ * Duke can also list the existing tasks and mark tasks as done.
+ */
+
+@SuppressWarnings("checkstyle:Regexp")
+public class Duke {
+
+    private int index = 0; //will figure out a way to make this private :<
+    private final String filePath;
+    private final TaskList tasks;
+    private Ui ui;
+
+    /**
+     * Constructor for the Duke object.
+     *
+     * @param filePath
+     */
+
+    public Duke(String filePath) {
+        this.filePath = filePath;
+        Storage store = new Storage(filePath, this);
+        tasks = new TaskList(store.loadTasks());
+        this.ui = new Ui();
+    }
+
+    public int getIndex() {
+        return this.index;
+    }
+
+    public void setIndex(int index) {
+        this.index = index;
+    }
+
+    /**
+     * Main function that does the main flow of the application
+     *
+     * @param args
+     */
+
+    public static void main(String[] args) {
+
+        Duke bot = new Duke("data/Duke.txt");
+        Scanner scanObj = new Scanner(System.in);
+
+        bot.ui.printGreeting();
+
+        while (true) {
```

Comment:
Maybe can extract this out into a run() method? like suggested in the weekly task page.